
syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.pubref.rules_protobuf.examples.thinkerer";
option java_outer_classname = "ThinkererProto";

import "proto/common.proto";

package thinkerer;

// The greeting service definition.
service Thinkerer {
  rpc SendMessage (Msg) returns (MsgReply) {}
  rpc RecvMessages (MsgReq) returns (Msgs) {}
}

message Msg {
  string from = 1;
  string to = 2;
  string message = 3;
}

message MsgReq {
  string uid = 1;
}

message Msgs {
  repeated Msg messages = 1;
}

message MsgReply {
  int32 status = 1;
}
